crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -31,21 +33,24 @@ fan_in_message_type!(MqttRuntimeMessage[MqttMessage, RuntimeRequest] : Debug, Cl
 
 pub struct MqttActorBuilder {
     pub mqtt_config: mqtt_channel::Config,
-    pub publish_channel: (Sender<MqttRuntimeMessage>, Receiver<MqttRuntimeMessage>),
+    input_receiver: CombinedReceiver<MqttRuntimeMessage>,
+    publish_sender: Sender<MqttRuntimeMessage>,
     pub subscriber_addresses: Vec<(TopicFilter, DynSender<MqttMessage>)>,
     signal_sender: Sender<RuntimeRequest>,
-    signal_receiver: Receiver<RuntimeRequest>,
 }
 
 impl MqttActorBuilder {
     pub fn new(config: mqtt_channel::Config) -> Self {
+        let (publish_sender, publish_receiver) = channel(10);
         let (signal_sender, signal_receiver) = channel(10);
+        let input_receiver = CombinedReceiver::new(publish_receiver, signal_receiver);
```

Comment:
An improvement could be to add a `CombinedSender` type along a `chanel()` that create a sender, receiver pair.

This combined sender will have to `impl RuntimeRequestSink` to ease the implementation of specific actor builders.